src/prometheus-net.DotNetRuntime.Tests/StatsCollectors/IntegrationTests/ExceptionStatsCollectorTests.cs

```diff
@@ -0,0 +1,67 @@
+using NUnit.Framework;
+using Prometheus.DotNetRuntime.StatsCollectors;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Prometheus.DotNetRuntime.Tests.StatsCollectors.IntegrationTests
+{
+    [TestFixture]
+    internal class ExceptionStatsCollectorTests : StatsCollectorIntegrationTestBase<ExceptionStatsCollector>
+    {
+        protected override ExceptionStatsCollector CreateStatsCollector()
+        {
+            return new ExceptionStatsCollector();
+        }
+
+        [Test]
+        public void Will_measure_when_occurring_an_exception()
+        {
+            // arrange
+            int divider = 0;
+            string exceptionMessage = string.Empty;
+
+            // act
+            try
+            {
+                var result = 1 / divider;
+            }
+            catch (Exception ex)
+            {
+                exceptionMessage = ex.Message;
+            }
+
+            // throwing exceptions are slower so please wait 20 ms.
+            Thread.Sleep(20);
```

Comment:
Rather than using `Thread.Sleep`, it's preferable to use the polling functionality built in to NUnit, e.g:
```
// Wait for 1 second for condition to be true, check the value every 100ms for an update
 Assert.That(() => StatsCollector.ExceptionReasons.Labels(exceptionMessage).Value, Is.EqualTo(1).After(100, 1000));
```